src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                {
+                    return;
+                }
+
+                var guardMethods = GetRuntimePlatformGuardMethods(runtimeInformationType!);
+
+                context.RegisterOperationBlockStartAction(context => AnalyzerOperationBlock(context, guardMethods, osPlatformType));
+            });
+
+            static ImmutableArray<IMethodSymbol> GetRuntimePlatformGuardMethods(INamedTypeSymbol runtimeInformationType)
+            {
+                return runtimeInformationType.GetMembers().OfType<IMethodSymbol>().Where(m =>
+                    s_platformCheckMethodNames.Contains(m.Name) && !m.Parameters.IsEmpty).ToImmutableArray();
+            }
+        }
+
+        private void AnalyzerOperationBlock(OperationBlockStartAnalysisContext context, ImmutableArray<IMethodSymbol> guardMethods, INamedTypeSymbol osPlatformType)
+        {
+            var platformSpecificOperations = PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>>.GetInstance();
+
+            context.RegisterOperationAction(context =>
+            {
+                AnalyzeInvocationOperation(context.Operation, context, ref platformSpecificOperations);
+            }
+            , OperationKind.DelegateCreation, OperationKind.EventReference, OperationKind.FieldReference,
+            OperationKind.Invocation, OperationKind.ObjectCreation, OperationKind.PropertyReference);
+
+            context.RegisterOperationBlockEndAction(context =>
+            {
+                try
+                {
+                    if (platformSpecificOperations.Count == 0)
+                    {
+                        return;
+                    }
+
+                    if (guardMethods.IsEmpty || !(context.OperationBlocks.GetControlFlowGraph() is { } cfg))
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
+                        cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
+                        context.Options, MinimumOsRule, performValueContentAnalysis: true,
+                        context.CancellationToken, out var valueContentAnalysisResult);
+
+                    if (analysisResult == null)
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    foreach (var platformSpecificOperation in platformSpecificOperations)
+                    {
+                        var value = analysisResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                        PlatformAttributeInfo attribute = platformSpecificOperation.Value.FirstOrDefault();
```

Comment:
Hmm, this seems wrong. If a method specifies that it is applicable to 2 platforms with separate attributes, we will ignore all but first in our analysis. Do we have tests to validate multi-attribute cases?

Reply:
It seems like the correct approach would be for `IsKnownValueGuarded` to take all attributes and return true only if _all_ platforms are guarded.

Reply:
Possibly write this as `ReportDiagnosticsForNotGuardedAttributes(platformSpecificOperation.Key, value, attributes, context)` and invoke this only for values where `value.Kind == GlobalFlowStateAnalysisValueSetKind.Known`.

Reply:
> If a method specifies that it is applicable to two or more platforms with separate attributes, we will ignore all but first in our analysis. Do we have tests to validate multi-attribute cases

I am aware of that, the initial design was for supporting only one attribute, but there was a possibility to of need to support multiple, so i have used array as most likely we will need to support many, And with the latest design we gonna support multiple attributes so it will be changed and corresponding tests will be added